xarray/backends/api.py

```diff
@@ -79,6 +83,42 @@ def check_name(name):
         check_name(k)
 
 
+def _validate_attrs(dataset):
+    """`attrs` must have a string key and a value which is either: a number
+    a string, an ndarray or a list/tuple of numbers/strings.
+    """
+    def check_attr(name, value):
+        if isinstance(name, basestring):
+            if not name:
+                raise ValueError('Invalid name for attr: string must be length '
+                                 '1 or greater for serialization to netCDF '
+                                 'files')
```

Comment:
Prefer to print invalid values in error messages (it helps for debugging), e.g., 
```
raise ValueError('Invalid value: {}'.format(value))
```